js/packages/candy-machine-ui/src/Home.tsx

```diff
@@ -0,0 +1,248 @@
+import { useWallet } from '@solana/wallet-adapter-react';
+import { WalletDialogButton } from '@solana/wallet-adapter-material-ui';
+import {
+  awaitTransactionSignatureConfirmation,
+  CandyMachineAccount,
+  CANDY_MACHINE_PROGRAM,
+  getCandyMachineState,
+  mintOneToken,
+} from './candy-machine';
+import { AlertState } from './utils';
+import { Header } from './Header';
+import { MintButton } from './MintButton';
+import { GatewayProvider } from '@civic/solana-gateway-react';
+
+const ConnectButton = styled(WalletDialogButton)`
+  width: 100%;
+  height: 60px;
+  margin-top: 10px;
+  margin-bottom: 5px;
+  background: linear-gradient(180deg, #604ae5 0%, #813eee 100%);
+  color: white;
+  font-size: 16px;
+  font-weight: bold;
+`;
+
+const MintContainer = styled.div``; // add your owns styles here
+
+export interface HomeProps {
+  candyMachineId?: anchor.web3.PublicKey;
+  connection: anchor.web3.Connection;
+  startDate: number;
+  txTimeout: number;
+  rpcHost: string;
+}
+
+const Home = (props: HomeProps) => {
+  const [isUserMinting, setIsUserMinting] = useState(false);
+  const [candyMachine, setCandyMachine] = useState<CandyMachineAccount>();
+  const [alertState, setAlertState] = useState<AlertState>({
+    open: false,
+    message: '',
+    severity: undefined,
+  });
+
+  const rpcUrl = props.rpcHost;
+  const wallet = useWallet();
+
+  const anchorWallet = useMemo(() => {
+    if (
+      !wallet ||
+      !wallet.publicKey ||
+      !wallet.signAllTransactions ||
+      !wallet.signTransaction
+    ) {
+      return;
+    }
+
+    return {
+      publicKey: wallet.publicKey,
+      signAllTransactions: wallet.signAllTransactions,
+      signTransaction: wallet.signTransaction,
+    } as anchor.Wallet;
+  }, [wallet]);
+
+  const refreshCandyMachineState = useCallback(() => {
+    (async () => {
+      if (!anchorWallet) {
+        return;
+      }
+
+      if (props.candyMachineId) {
+        try {
+          const cndy = await getCandyMachineState(
+            anchorWallet,
+            props.candyMachineId,
+            props.connection,
+          );
+          setCandyMachine(cndy);
+        } catch (e) {
+          console.log('There was a problem fetching Candy Machine state');
+          console.log(e);
+        }
+      }
+    })();
+  }, [anchorWallet, props.candyMachineId, props.connection]);
+
+  const onMint = async () => {
+    try {
+      setIsUserMinting(true);
+      document.getElementById('#identity')?.click();
+      if (wallet.connected && candyMachine?.program && wallet.publicKey) {
+        const mintTxId = (
+          await mintOneToken(candyMachine, wallet.publicKey)
+        )[0];
+
+        let status: any = { err: true };
+        if (mintTxId) {
+          status = await awaitTransactionSignatureConfirmation(
+            mintTxId,
+            props.txTimeout,
+            props.connection,
+            true,
+          );
+        }
+
+        if (!status?.err) {
+          setAlertState({
```

Comment:
`status` could [resolve to `null`](https://github.com/metaplex-foundation/metaplex/pull/1424/files#diff-e3ddb2a90d60124d177d58ff771036b73703706eb73d361ce8467fcf4232857eL63) which would cause this success message to be set. is this intended?